Added option for exceptions

magnuswatn opened the pull request:
Hi,

I added the option for exceptions. It's simple and the GUI is rudimentary, but it works :-)

magnuswatn committed:
Added option for exceptions

lgarron commented:
Thanks for implementing!

I haven't tried it yet, but the basic idea sounds fine. I'll leave comments on the code.

lgarron commented on content.js:
```diff
@@ -17,14 +17,18 @@ function defaultTheme() {
+    exceptions: ""
   }, function(items) {
     var theme = items.theme;
+    var exceptions = items.exceptions.split(',')
```
However, I'm a little concerned about parsing an exception list on every page load. I don't think you can store `Set`s in `chrome.storage`, but could you store it as a map?

    {
      "bing.com": true
      "example.com": true
    }

lgarron commented on content.js:
```diff
@@ -17,14 +17,18 @@ function defaultTheme() {
     }
-
-    document.body.classList.add("ugly-http-theme-" + theme);
+    if (!exceptions.includes(window.location.host)) {
```
`host` can include an optional port. Could you use `hostname` instead?

lgarron commented on options/options.html:
```diff
@@ -10,28 +10,53 @@
+          Exceptions:
+        </td>
+        <td>
+          <input type="text" id="exceptions">
```
Could we make this a `textarea` with newlines between hosts?

magnuswatn commented:
Thanks for the feedback! I will have a look at this again tomorrow.

magnuswatn committed:
Rewrote so that the exception list is stored as a map

Also changed the input to textarea and changed so that the exception covers the whole host (not only a specific port)

magnuswatn commented:
Sorry this took so long, time flew by... I had a go again, it would be great if you could take a look!

lgarron commented on options/options.js:
```diff
@@ -10,20 +12,45 @@ function display_status(message) {
     document.getElementById("theme").value = items.theme;
+    var exceptionList = document.getElementById("exceptions");
+    for (var exception in items.exceptions) {
+        exceptionList.value += exception + '\n';
```
Does the trailing newline cause an empty entry when serializing, or is that avoided?

magnuswatn replied:
That is avoided by:
`if (item)` (on line 20)
An empty string will be false, and thus is not added.

lgarron commented on options/options.js:
```diff
@@ -10,20 +12,45 @@ function display_status(message) {
   chrome.storage.local.get({
-    theme: "default"
+    theme: "default",
+    exceptions: ""
```
(same as above)

lgarron commented on content.js:
```diff
@@ -17,14 +17,17 @@ function defaultTheme() {
   chrome.storage.local.get({
-    theme: "default"
+    theme: "default",
+    exceptions: ""
```
This should be `{}` now, right?

magnuswatn replied:
You are, of course, absolutely correct. My bad. I have fixed this now.

lgarron reviewed (changes requested):
Thanks! I've left a few more comments about details.

magnuswatn committed:
Corrected default value for exceptions when retrieving from chrome.store

(it is now a map, not a string)

lgarron commented:
Merged and released. Thanks!
